### PR TITLE
Fix urql fetching and caching in the app

### PR DESCRIPTION
**Pull request by orzechdev**
This adds:
- fix of unintended caching of settings in API endpoints and consequently returning legacy settings in all endpoints
- fix of missing authorized requests on opening the app and consequently e.g. missing channels on channels page
- preventing to make unauthorized requests when token is not available, as it should make them only after obtaining it from app-bridge on client side

### CODE REVIEW

File: apps/checkout-app/frontend/components/elements/AppProvider/AppProvider.tsx [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
 }
 
-export const AppContext = createContext<IAppContext>({ app: undefined });
+export const AppContext = createContext<IAppContext>({
+  app: undefined,
+  token: undefined,
+});
 
 const AppProvider: React.FC = (props) => {
+  const [token, setToken] = useState<string>();
```
Comment (jwm0) [inline apps/checkout-app/frontend/components/elements/AppProvider/AppProvider.tsx:16]: Token should not be kept in React state, instead you can call `createApp` outside the component and use `app.getState().token` inside client

File: apps/checkout-app/frontend/hooks/useAuthData.ts [hunk 2]
```diff
@@ -3,12 +3,17 @@ import { useApp } from "./useApp";
...
   app: string;
+  isAuthorized: boolean;
 }
 
 export const useAuthData = (): AuthTokenPayload => {
   const app = useApp();
-  const appState = app?.getState();
-  const payload = appState?.token ? parseJwt(appState.token) : { app: "" };
+  const appState = app.app?.getState();
+  const payload = appState?.token ? parseJwt(appState.token) : {};
```
Comment (jwm0) [inline apps/checkout-app/frontend/hooks/useAuthData.ts:12]: appId is now passed in from app-bridge

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: apps/checkout-app/frontend/components/elements/ClientProvider/ClientProvider.tsx [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
 const ClientProvider: React.FC = ({ children, ...props }) => {
-  const app = useApp();
+  const { app } = useApp();
 
   const token = app?.getState()?.token;
 
-  const client = useMemo(() => {
-    if (token) {
-      return getClient(token);
+  const client = useRef(getClient(API_URL, token));
```
Comment (jwm0) [inline apps/checkout-app/frontend/components/elements/ClientProvider/ClientProvider.tsx:12]: The token should not be passed from a React view but rather the `app` should be imported and this:
```
      getAuth: getAuth(token),
```
should be:
```
      getAuth: () => app.getState().token,
```

**Review by bmigirl — APPROVED** [review]

File: apps/checkout-app/frontend/components/elements/ClientProvider/ClientProvider.tsx (hunk 3 above)
Comment (orzechdev) [reply apps/checkout-app/frontend/components/elements/ClientProvider/ClientProvider.tsx]: I've changed it to passing app to getClient and getting token as you suggested in getAuth

**Review by jwm0 — APPROVED** [review]

**Pull request merged**